bodhi/server/consumers/signed.py

```diff
@@ -110,7 +110,7 @@ def __call__(self, message: fedora_messaging.api.Message):
             log.info("Build %s has been marked as signed" % build_nvr)
 
             # If every build in update is signed change status to testing
-            if build.update.from_tag and build.update.signed():
+            if not build.update.release.composed_by_bodhi and build.update.signed():
                 log.info("Every build in update is signed, set status to testing")
```

Comment:
I think we should call greenwave to see if the update should remain in testing or not here.

Basically: move the update from Pending to Testing, check with greenwave immediately if the update should be gated and it not directly move it to Stable.

Reply:
Cf the lack deleted block in ``automatic_updates.py``

Reply:
> I think we should call greenwave to see if the update should remain in testing or not here.
> 
> Basically: move the update from Pending to Testing, check with greenwave immediately if the update should be gated and it not directly move it to Stable.

Even if the update is no gated don't we want to wait for the tests to finishing running before moving to stable ? That way we get the same experience if an update is gated or not ( ie one will not be faster than the other)

Reply:
> Even if the update is no gated don't we want to wait for the tests to finishing running before moving to stable ? 

This assumes there will be test running which is currently not the case most often :)

If we want to minimize the impact of gating and I think if we do not want to gate a build if there is no reasons to :)

Reply:
> > Even if the update is no gated don't we want to wait for the tests to finishing running before moving to stable ?
> 
> This assumes there will be test running which is currently not the case most often :)

Ok so if I read the code correctly then we don't need to call greenwave in that case, since by default update.test_gating_status is None and if the status is None we consider the tests passed (see code below). Which means that in the update would be pushed to Stable the next time the cronjob runs.

https://github.com/fedora-infra/bodhi/blob/develop/bodhi/server/models.py#L2449

> 
> If we want to minimize the impact of gating and I think if we do not want to gate a build if there is no reasons to :)

Reply:
> > > Even if the update is no gated don't we want to wait for the tests to finishing running before moving to stable ?
> > 
> > 
> > This assumes there will be test running which is currently not the case most often :)
> 
> Ok so if I read the code correctly then we don't need to call greenwave in that case, since by default update.test_gating_status is None and if the status is None we consider the tests passed (see code below). Which means that the update would be pushed to Stable the next time the cronjob runs.

Hum this can then be a problem if the update has tests and is gated, since the test_gating_status will be None. So I guess there is no way around having to call greenwave at that point.
> 
> https://github.com/fedora-infra/bodhi/blob/develop/bodhi/server/models.py#L2449
> 
> > If we want to minimize the impact of gating and I think if we do not want to gate a build if there is no reasons to :)